crates/fuel-core/src/executor.rs

```diff
@@ -908,29 +903,17 @@ where
                     maturity,
                     ..
                 } => {
-                    let block_created = if utxo_validation {
-                        db.storage::<Coins>()
-                            .get(utxo_id)?
-                            .ok_or(ExecutorError::TransactionValidity(
-                                TransactionValidityError::CoinDoesNotExist(*utxo_id),
-                            ))?
-                            .block_created
-                    } else {
-                        // if utxo validation is disabled, just assign this new input to the original block
-                        Default::default()
-                    };
-
-                    db.storage::<Coins>().insert(
-                        utxo_id,
-                        &CompressedCoin {
-                            owner: *owner,
-                            amount: *amount,
-                            asset_id: *asset_id,
-                            maturity: (*maturity).into(),
-                            status: CoinStatus::Spent,
-                            block_created,
-                        },
+                    let mut coin = self.get_coin_or_default(
+                        db,
+                        *utxo_id,
+                        *owner,
+                        *amount,
+                        *asset_id,
+                        (*maturity).into(),
                     )?;
+                    coin.status = CoinStatus::Spent;
```

Comment:
It is not related to this PR, but maybe we want to do the same as we did for the message - a separate column to track whether is coin spent or not.